src/applications/app.helpers.js

```diff
@@ -36,15 +36,53 @@ export function isntLoaded(app) {
 
 export function shouldBeActive(app) {
   try {
-    return app.activeWhen(window.location);
+    if (Array.isArray(app.activeWhen))
+      return app.activeWhen.some(conditionForAppActivationMet);
+    return conditionForAppActivationMet(app.activeWhen);
   } catch (err) {
     handleAppError(err, app, SKIP_BECAUSE_BROKEN);
   }
+
+  function conditionForAppActivationMet(pathPrefixOrActivityFn) {
+    const { location } = window;
+    if (typeof pathPrefixOrActivityFn === "string") {
+      const hashRouting = pathPrefixOrActivityFn.startsWith("/#/");
+      const [
+        dynamicIndexes,
+        pathPrefixWithoutDynamic,
+      ] = pathPrefixOrActivityFn.split("/").reduce(
+        ([dynamicIndexes, pathPrefixWithoutDynamic], path, index) => {
+          if (path.includes(":")) {
+            return [
+              dynamicIndexes.concat(index),
+              pathPrefixWithoutDynamic.concat(""),
+            ];
+          }
+
+          return [dynamicIndexes, pathPrefixWithoutDynamic.concat(path)];
+        },
+        [[], []]
+      );
+      const currentPath = hashRouting ? `/${location.hash}` : location.pathname;
+      const replaceDynamicWitEmptyString = (path) =>
+        path
+          .split("/")
+          .map((item, index) => (dynamicIndexes.includes(index) ? "" : item))
+          .join("/");
+      return replaceDynamicWitEmptyString(currentPath).startsWith(
```

Comment:
typo :'(

Reply:
Also here, I could use the same function `replaceDynamicWitEmptyString` to remove dynamic entries on the `pathPrefixOrActivityFn` string, but I preferred to use the iteration in the first `reduce`... maybe its a premature optimization?